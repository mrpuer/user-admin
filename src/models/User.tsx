import { types } from "mobx-state-tree";

const User = types.model({
  age: types.optional(types.number, 0),
  email: "",
  name: types.string,
});

export default User;
