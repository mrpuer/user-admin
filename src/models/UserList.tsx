import { types } from "mobx-state-tree";
import User from "./User";

const UserList = types.model({
  users: types.optional(types.array(User), []),
}).actions((self) => ({
  add(item: any) {
    self.users.push(item);
  },
}));

export default UserList;
