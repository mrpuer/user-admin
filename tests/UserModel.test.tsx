import { getSnapshot } from "mobx-state-tree";

import User from "../src/models/User";
import UserList from "../src/models/UserList";

it("Can create a instance of a User", () => {
  const item = User.create({
    age: 30,
    name: "TestUser",
  });

  expect(item.age).toBe(30);
  expect(item.email).toBe("");
});

it("Can create a UserList", () => {
  const item = UserList.create({
    users: [{
      age: 30,
      name: "TestUser",
    }],
  });

  const emptyItem = UserList.create({});

  expect(emptyItem.users.length).toBe(0);
  expect(item.users.length).toBe(1);
  expect(item.users[0].age).toBe(30);
  expect(item.users[0].email).toBe("");
});

it("Can add user to UserList", () => {
  const item = UserList.create({});
  item.add({
    age: 30,
    name: "TestUser",
  });

  expect(item.users.length).toBe(1);
  expect(item.users[0].age).toBe(30);
  expect(item.users[0].email).toBe("");

  expect(getSnapshot(item)).toMatchSnapshot();
});
