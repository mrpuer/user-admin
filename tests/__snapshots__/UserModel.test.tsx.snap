// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Can add user to UserList 1`] = `
Object {
  "users": Array [
    Object {
      "age": 30,
      "email": "",
      "name": "TestUser",
    },
  ],
}
`;
